app/controllers/candidate_interface/gcse/science/grade_controller.rb

```diff
@@ -1,39 +1,108 @@
+      @gcse_grade_form = science_gcse_grade_form
+      @qualification_type = gcse_science_qualification.qualification_type
+
+      render view_path
     end
 
     def update
-      @qualification_type = details_form.qualification.qualification_type
-
-      details_form.grade = details_params[:grade]
-      details_form.other_grade = details_params[:other_grade]
-
-      @application_qualification = details_form.save_grade
+      @gcse_grade_form = science_gcse_grade_form.tap do |form|
+        form.subject = science_details_params[:gcse_science] || @subject
+        form.grade = assign_grade
+        form.biology_grade = science_details_params[:biology_grade]
+        form.physics_grade = science_details_params[:physics_grade]
+        form.chemistry_grade = science_details_params[:chemistry_grade]
+        form.gcse_science = science_details_params[:gcse_science]
+      end
 
-      if @application_qualification
+      if @gcse_grade_form.save_grades
         update_gcse_completed(false)
-        redirect_to next_gcse_path
+
+        redirect_to next_path
       else
-        @application_qualification = details_form
-        track_validation_error(@application_qualification)
+        track_validation_error(@gcse_grade_form)
 
-        render :edit
+        render view_path
       end
     end
 
   private
 
-    # Required by the GradeControllerConcern
+    def view_path
+      if gcse_qualification? && FeatureFlag.active?(:science_gcse_awards)
+        'candidate_interface/gcse/science/grade/awards_edit'
+      else
+        'candidate_interface/gcse/science/grade/edit'
+      end
+    end
+
+    def next_path
+      if science_gcse_grade_form.award_year.nil?
+        candidate_interface_gcse_details_edit_year_path(subject: @subject)
+      else
+        candidate_interface_gcse_review_path(subject: @subject)
+      end
+    end
+
+    def gcse_qualification?
+      gcse_science_qualification.qualification_type == 'gcse'
+    end
+
+    def single_award?
+      science_details_params[:gcse_science] == ApplicationQualification::SCIENCE_SINGLE_AWARD
+    end
+
+    def double_award?
+      science_details_params[:gcse_science] == ApplicationQualification::SCIENCE_DOUBLE_AWARD
+    end
+
+    def triple_award?
+      science_details_params[:gcse_science] == ApplicationQualification::SCIENCE_TRIPLE_AWARD
+    end
+
     def set_subject
       @subject = 'science'
     end
+
+    def science_details_params
+      params.require(:candidate_interface_science_gcse_grade_form)
+            .permit(%i[gcse_science grade single_award_grade double_award_grade biology_grade chemistry_grade physics_grade])
+    end
+
+    def assign_grade
+      if single_award?
+        science_details_params[:single_award_grade]
+      elsif double_award?
+        science_details_params[:double_award_grade]
+      else
+        science_details_params[:grade]
+      end
+    end
+
+    def assign_subject
```

Comment:
Is this being used anywhere? It looks like the build_from_qualification method uses a case statement with the same logic inside the form.

Reply:
Yep that's old code that I forgot to delete - will delete